sklearn/utils/stats.py

```diff
@@ -41,6 +41,8 @@ def _weighted_percentile(array, sample_weight, percentile=50):
     sorted_weights = np.take_along_axis(sample_weight, sorted_idx, axis=0)
 
     # Find index of median prediction for each sample
+    # nan_mask = np.isnan(array) #???
+    # weight_cdf[nan_mask] = 0 #
```

Comment:
Yes, this will eliminate the issue that varying weights for values that are nans could influence `_weighted_percentile` to deliver a different result.

> Note that when the weights are already passed as a 2d array, we should be careful not to mutate the user provided input and instead make a copy of the weights before setting some of the entries to zero (I think).

I decided to set the weights from nan-values to 0 on `sorted_weights` not on `sample_weight`) and I believe that I don't need to copy thus, correct?